engine/src/main/java/org/hibernate/search/bridge/util/impl/NumericFieldUtils.java

```diff
@@ -24,6 +41,9 @@
 
 	private static final Log log = LoggerFactory.make();
 
+	private static final Class<?>[] NUMERIC_JAVA_TIME_TYPES = { Year.class, Instant.class, LocalDate.class, LocalDateTime.class, YearMonth.class,
```

Comment:
Does that work on JDKs < 8? I'd assume the class load + static initialisation will stumble upon these types here on older JDKs?